test/hotspot/jtreg/runtime/ClassInitErrors/InitExceptionUnloadTest.java

```diff
@@ -0,0 +1,137 @@
+            throw new RuntimeException(expected + " and " + cause + " missing from stacktrace");
+        }
+    }
+
+    static String[] expected = new String[] {
+        "java.lang.ExceptionInInitializerError",
+        "Caused by: java.lang.ArithmeticException: / by zero",
+        "java.lang.NoClassDefFoundError: Cound not initialize class InitExceptionUnloadTest$ThrowsRuntimeException",
+        "Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.ArithmeticException: / by zero [in thread",
+        "java.lang.Error",
+        null,
+        "java.lang.NoClassDefFoundError: Cound not initialize class InitExceptionUnloadTest$ThrowsError",
+        "Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.Error [in thread",
+        "java.lang.ExceptionInInitializerError",
+        "Caused by: InitExceptionUnloadTest$SpecialException: Very Special 3",
+        "java.lang.NoClassDefFoundError: Cound not initialize class InitExceptionUnloadTest$ThrowsSpecialException",
+        "Caused by: java.lang.ExceptionInInitializerError: Exception InitExceptionUnloadTest$SpecialException: Very Special 3",
+        "java.lang.OutOfMemoryError",
+        "Java heap space",
+        "java.lang.NoClassDefFoundError: Cound not initialize class InitExceptionUnloadTest$InitOOM",
+        "Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.OutOfMemoryError: Java heap space [in thread"
+    };
+
+    static String[] classNames = new String[] {
+        "InitExceptionUnloadTest$ThrowsRuntimeException",
+        "InitExceptionUnloadTest$ThrowsError",
+        "InitExceptionUnloadTest$ThrowsSpecialException",
+        "InitExceptionUnloadTest$InitOOM" };
+
+    public static WhiteBox wb = WhiteBox.getWhiteBox();
+
+    static void test() throws Throwable {
+        ClassLoader cl = ClassUnloadCommon.newClassLoader();
+        int i = 0;
+        for (String className : classNames) {
+            for (int tries = 2; tries--> 0; ) {
+                System.err.println("--- try to load " + className);
+                try {
+                    Class<?> c = cl.loadClass(className);
+                    Object inst = c.newInstance();
+                } catch (Throwable t) {
+                    t.printStackTrace();
+                    System.err.println();
+                    System.err.println("Check results");
+                    verify_stack(t, expected[i], expected[i+1]);
+                    i += 2;
+                    System.err.println();
+                }
+            }
+        }
+        cl = null;
+        ClassUnloadCommon.triggerUnloading();  // should unload these classes
+        for (String className : classNames) {
+          ClassUnloadCommon.failIf(wb.isClassAlive(className), "should be unloaded");
+        }
+    }
+    public static void main(java.lang.String[] unused) throws Throwable {
+        test();
+        test();
```

Comment:
Why twice?

Reply:
The classes should be unloaded and the message should be the same for each iteration, ie, first call should get EIIE and second should get NCDFE.